### PR TITLE
DECO-91 - Container status

### PR DESCRIPTION
**Pull request by DennisOfTheBurn**
Implementation of the retrieval of the modem firmware version and addition of float support for the JSON encoder in order to generate the correct container status JSON to send to the AWS shadow.

### CODE REVIEW

File: drivers/modem/gsm_ppp.c [hunk 1]
```diff
@@ -709,6 +724,9 @@ static const struct setup_cmd setup_modem_info_cmds[] = {
 	SETUP_CMD("AT+CIMI", "", on_cmd_atcmdinfo_imsi, 0U, ""),
 	SETUP_CMD("AT+CCID", "", on_cmd_atcmdinfo_iccid, 0U, ""),
 #endif
+#if defined (CONFIG_MODEM_FIRMWARE_VERSION)
+	SETUP_CMD("AT+QGMR", "", on_cmd_atcmdinfo_firmware, 0U, ""),
```
Comment (n-nievergeld) [inline drivers/modem/gsm_ppp.c:728]: Note that this is probably a quectel specific command, so maybe check for MODEM_QUECTEL as well.

Comment (n-nievergeld) [reply drivers/modem/gsm_ppp.c:728]: Also check other `#if defined (CONFIG_MODEM_FIRMWARE_VERSION)`

File: lib/os/json.c [hunk 2]
```diff
@@ -989,12 +1006,14 @@ static int encode(const struct json_obj_descr *descr, const void *val,
 				       ptr, append_bytes, data);
 	case JSON_TOK_NUMBER:
 		return num_encode(ptr, append_bytes, data);
-	case JSON_TOK_NULL:
-		return null_encode(ptr, append_bytes, data);
 	case JSON_TOK_FLOAT:
+		return float_encode(ptr, append_bytes, data);
+	case JSON_TOK_FLOAT_ASCII:
```
Comment (n-nievergeld) [inline lib/os/json.c:1011]: This does not seem to be compatilbe. Are you sure no other code in zephyr does rely on `JSON_TOK_FLOAT` actually doing a `float_ascii_encode`?

Comment (n-nievergeld) [reply lib/os/json.c:1011]: Or is this only be used internally?

**Review by n-nievergeld — APPROVED** [review]
Few comments, but I already approved

File: lib/os/json.c (hunk 2 above)
Comment (DennisOfTheBurn) [reply lib/os/json.c:1011]: Yes, this is from the first change into the Zephyr v3.1 json library, which did not support floats yet. I will improve on it to be inline with the Zephyr v3.2 json library.

File: drivers/modem/gsm_ppp.c (hunk 1 above)
Comment (DennisOfTheBurn) [reply drivers/modem/gsm_ppp.c:730]: Fixed by making `MODEM_FIRMWARE_VERSION` depended on `MODEM_GSM_QUECTEL` in the Kconfig file.

**Pull request merged**